query-grammar/src/lib.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::derive_partial_eq_without_eq)]
```

Comment:
Does it really make sense to allow this instead of just adding the derive or is this mainly about the code noise of doing that?

Reply:
The lint feels pointless to me.